### PR TITLE
test: add integration test for task auto recover

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
### What problem does this PR solve? 
 add integration test for task auto recover

### What is changed and how it works?
add a test in `all_mode`

### Check List 

Tests 

 - Integration test

### CODE REVIEW

File: tests/all_mode/run.sh [hunk 1]
```diff
@@ -32,14 +34,29 @@ function run() {
...
     run_sql_file $cur/data/db2.increment.sql $MYSQL_HOST2 $MYSQL_PORT2
 
+    sleep 2
+    # dm-worker execute sql failed, and will try auto resume task
+    check_log_contains $WORK_DIR/worker1/log/dm-worker.log "dispatch auto resume task"
+    check_log_contains $WORK_DIR/worker2/log/dm-worker.log "dispatch auto resume task"
+
+    # restart tidb, and task will recover success
+    run_tidb_server 4000
+    sleep 1
```
Comment (csuzhangxc) [inline tests/all_mode/run.sh]: Is it possible more time needed to wait?

Comment (WangXiangUSTC) [reply tests/all_mode/run.sh]: check_sync_diff will retry several times, so it doesn't matter. and I increase to 2

File: dm/worker/task_checker.go [hunk 2]
```diff
@@ -203,6 +204,17 @@ func (tsc *realTaskStatusChecker) Close() {
 func (tsc *realTaskStatusChecker) run() {
 	tsc.ctx, tsc.cancel = context.WithCancel(context.Background())
 	tsc.closed.Set(closedFalse)
+
+	failpoint.Inject("TaskCheckInterval", func(val failpoint.Value) {
+		interval, ok := val.(int)
+		if !ok {
+			tsc.l.Warn("inject failpoint failed", zap.Reflect("value", val))
+		} else {
+			tsc.cfg.CheckInterval = time.Duration(interval) * time.Millisecond
```
[the remark this answers is not in the document]
Comment (WangXiangUSTC) [reply dm/worker/task_checker.go]: update

File: dm/worker/task_checker.go [hunk 3]
```diff
@@ -203,6 +204,17 @@ func (tsc *realTaskStatusChecker) Close() {
 func (tsc *realTaskStatusChecker) run() {
 	tsc.ctx, tsc.cancel = context.WithCancel(context.Background())
 	tsc.closed.Set(closedFalse)
+
+	failpoint.Inject("TaskCheckInterval", func(val failpoint.Value) {
+		interval, ok := val.(int)
+		if !ok {
+			tsc.l.Warn("inject failpoint failed", zap.Reflect("value", val))
```
[the remark this answers is not in the document]
Comment (WangXiangUSTC) [reply dm/worker/task_checker.go]: done

**Comment by csuzhangxc** [comment]
@amyangfei PTAL

**Pull request merged**